src/threading/MsgThread.cc

```diff
@@ -413,3 +445,24 @@ void MsgThread::GetStats(Stats* stats)
 	queue_out.GetStats(&stats->queue_out_stats);
 	}
 
+void MsgThread::Process()
+	{
+	// Read out all of the notification bytes since we're going to process all
+	// of the messages that are available.
+	char byte;
+	while ( read(ready_pair[0], &byte, 1) != -1 ) { }
```

Comment:
* Possibly more efficient to try reading a chunk of bytes out rather than one at a time.
* Maybe worth more explicit error checking of `errno` for the `read()` and `write()` calls: think generally we can expect to never see anything unusual there, but the kind of thing where if something goes wrong, the program state is messed up at a fundamental level and I'd at least like to see that reflected in a log.